### PR TITLE
Add type-defs autobind-decorator using typescript defs as a baseline

### PR DESCRIPTION
**Pull request by AVykhrystyuk**
- Links to documentation:  https://github.com/andreypopp/autobind-decorator
- Link to GitHub or NPM: https://github.com/andreypopp/autobind-decorator
- Type of contribution: new definition

Other notes:
Typescript typings - https://github.com/andreypopp/autobind-decorator/blob/master/index.d.ts

### CODE REVIEW

File: definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    declare export interface TypedPropertyDescriptor<T> {
+      enumerable?: boolean;
+      configurable?: boolean;
+      writable?: boolean;
+      value?: T;
+      get?: () => T;
+      set?: (value: T) => void;
+    }
+
+    declare export type ClassDecorator = <TFunction: Function>(target: TFunction) => TFunction;
```
Comment (pascalduez) [inline definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js]: `Function` is `any` as a Fow 0.75, so the constraint here will not work. Maybe `() => void`?

File: definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    declare export type MethodDecorator = <T>(target: Object, propertyKey: string | Symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T>;
```
Comment (pascalduez) [inline definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js]: Same for `Object`. `{}` should do it.

File: definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js (hunk 1 above)
Comment (AVykhrystyuk) [reply definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js]: Fixed 031d068e304411e6f4d0473b1f82336e8a36b51c

File: definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js (hunk 2 above)
Comment (AVykhrystyuk) [reply definitions/npm/autobind-decorator_v2.x.x/flow_v0.25.x-/autobind-decorator_v2.x.x.js]: Fixed cff1b4bda547fd8eb35c415d5ea4af4a8326bbc0

**Comment by AVykhrystyuk** [comment]
@pascalduez Thank you for a good review.
Looks like all your comments are taken care of

**Pull request merged**